tfhe/src/high_level_api/integers/tests.rs

```diff
@@ -147,6 +147,130 @@ fn test_uint32() {
     assert_eq!(decrypted, clear_a.wrapping_add(clear_b));
 }
 
+#[test]
+fn test_uint32_shift() {
+    let config = ConfigBuilder::all_disabled()
+        .enable_default_integers()
+        .build();
+
+    let (cks, sks) = generate_keys(config);
+
+    use rand::prelude::*;
+
+    let mut rng = rand::thread_rng();
+    let clear_a = rng.gen::<u32>();
+    let clear_b = rng.gen_range(0u32..32u32);
```

Comment:
no overflow case ? not sure what the semantics are for rust in that case

Reply:
semantics are same as masking shift value with the maximum num bits (shift % 32 in 32 bits), overflow values are tested in integer tests normally

Reply:
ah right, sorry forgot